diff --git a/devices/src/virtio/vhost_user_backend/gpu/sys/linux.rs b/devices/src/virtio/vhost_user_backend/gpu/sys/linux.rs
index a521680..db3ef6d 100644
--- a/devices/src/virtio/vhost_user_backend/gpu/sys/linux.rs
+++ b/devices/src/virtio/vhost_user_backend/gpu/sys/linux.rs
@@ -158,7 +158,7 @@ pub struct Options {
 
 pub fn run_gpu_device(opts: Options) -> anyhow::Result<()> {
     let Options {
-        x_display,
+        x_display: _,
         params: mut gpu_parameters,
         resource_bridge,
         socket,
@@ -222,13 +222,7 @@ pub fn run_gpu_device(opts: Options) -> anyhow::Result<()> {
 
     let (gpu_control_tube, _) = Tube::pair().context("failed to create gpu control tube")?;
 
-    let mut display_backends = vec![
-        virtio::DisplayBackend::X(x_display),
-        virtio::DisplayBackend::Stub,
-    ];
-    if let Some(p) = channels.get("") {
-        display_backends.insert(0, virtio::DisplayBackend::Wayland(Some(p.to_owned())));
-    }
+    let display_backends = vec![virtio::DisplayBackend::Stub];
 
     // These are only used when there is an input device.
     let event_devices = Vec::new();
diff --git a/src/crosvm/sys/linux/gpu.rs b/src/crosvm/sys/linux/gpu.rs
index 6c5b940..3ae0548 100644
--- a/src/crosvm/sys/linux/gpu.rs
+++ b/src/crosvm/sys/linux/gpu.rs
@@ -125,7 +125,6 @@ pub fn create_gpu_device(
         gpu_params.allow_implicit_render_server_exec && !is_sandboxed;
 
     let mut display_backends = vec![
-        virtio::DisplayBackend::X(cfg.x_display.clone()),
         virtio::DisplayBackend::Stub,
     ];
 
@@ -134,14 +133,6 @@ pub fn create_gpu_device(
         display_backends.insert(0, virtio::DisplayBackend::Android(service_name.to_string()));
     }
 
-    // Use the unnamed socket for GPU display screens.
-    if let Some(socket_path) = cfg.wayland_socket_paths.get("") {
-        display_backends.insert(
-            0,
-            virtio::DisplayBackend::Wayland(Some(socket_path.to_owned())),
-        );
-    }
-
     let dev = virtio::Gpu::new(
         exit_evt_wrtube
             .try_clone()
